plugins/widget/plugin.js

```diff
@@ -3399,19 +3418,34 @@
 			listeners = [],
 			sorted = [],
 			locations,
-			y;
+			y, x, target;
 
 		// Mark dragged widget for repository#finder.
 		this.repository._.draggedWidget = this;
 
+		var cellCount = CKEDITOR.tools.array.reduce( editable.find( 'tr' ).toArray(), function( total, element ) {
+					return total + element.getChildCount();
+				}, 0 ),
+			fakeParagraphHtml = '<p class="cke_fake-paragraph" style="height:0;margin:0;padding:0"></p>',
+			// To be adjusted. Higher value could be used for newer browsers and lower for older browsers.
+			limit = 100;
+
+		if ( cellCount <= limit ) {
+			var cells = editable.find( 'td' ).toArray().concat( editable.find( 'th' ).toArray() );
+			addFakeParagraphs( cells );
+		}
+
 		// Harvest all possible relations and display some closest.
 		var relations = finder.greedySearch(),
-
 			buffer = CKEDITOR.tools.eventsBuffer( 50, function() {
+				if ( cellCount > limit && target.getName() in { td:1, th:1 } ) {
+					storeFakeParagraph( addFakeParagraph( target ) );
```

Comment:
This two functions + two execution has only one purpose, faking drop spot. Maybe you could merge it into one function with additional helpers if needed?